src/ray/raylet/agent_manager.cc

```diff
@@ -98,31 +91,10 @@ void AgentManager::StartAgent() {
 
     int exit_code = child.Wait();
     timer->cancel();
-    RAY_LOG(WARNING) << "Agent process with pid " << child.GetId()
+    // Raylet shares fate with agent to make the failover logic easier.
+    RAY_CHECK(false) << "Agent process with pid " << child.GetId()
```

Comment:
Quick question, what's the difference between `RAY_CHECK(false)` and `RAY_LOG(ERROR)`?

Reply:
`RAY_CHECK(false)` will kill current process. Actually, we can also use `RAY_LOG(FATAL)` instead.